pkg/util/util.go

```diff
@@ -687,6 +687,11 @@ func HTTPGetRequest(url string) ([]byte, error) {
 	}
 	defer resp.Body.Close()
 
+	// we have a non 1xx / 2xx status, return an error
+	if (resp.StatusCode - 300) > 0 {
```

Comment:
consider 302? and follow redirects?

Reply:
The original code wasn't doing it, I'm just addressing #3123 here. If you feel a better implementation is needed (which might be a good idea), open an issue.